lib/cldr/export.rb

```diff
@@ -118,8 +118,15 @@ def locale_based_data(component, locale, options = {})
       end
 
       def plural_data(component, locale, options = {})
-        data = locale_based_data(component, locale, options)
-        data.empty? ? "" : "{ :'#{locale}' => { :i18n => { :plural => { :keys => #{data[:keys].inspect}, :rule => #{data[:rule]} } } } }"
+        result = locales(locale, component, options).lazy.map do |ancestor|
+          Data::Plurals.rules.slice(ancestor)
+        end.reject(&:empty?).first
+
+        if result && result.keys != [locale]
+          result[locale] = result.delete(result.keys.first)
+        end
```

Comment:
If we fell back, the rule data will have the locale of the ancestor.
This swaps it with the expected locale.